docs/flow/modin/experimental/core/execution/native/implementations/omnisci_on_native/index.rst

```diff
@@ -1,7 +1,237 @@
 :orphan:
 
+OmniSciOnNative execution
+=========================
+
+OmniSciDB is an open-source SQL-based relational database designed for the
+massive parallelism of modern CPU and GPU hardware. Its execution engine
+is built on LLVM JIT compiler.
+
+OmniSciDB can be embedded into an application as a dynamic library that
+provides both C++ and Python APIs. A specialized in-memory storage layer
+provides an efficient way to import data in Arrow table format.
+
+`OmniSciOnNative` execution uses OmniSciDB for both as a storage format and for
+actual data transformation.
+
+Relational engine limitations
+-----------------------------
+
+Using a relational database engine implies a set of restrictions on
+operations we can execute on a dataframe.
+
+1. We cannot handle frames that use data types not supported by OmniSciDB.
+Currently, we allow only integer, float, string, and categorical data types.
+
+2. Column data should be homogeneous.
+
+3. Can only support operations that map to relational algebra. This means
+most operations are supported over a single axis (axis=0) only. Non-relational
+operations like transposition and pivot are not supported.
+
+When the unsupported data type is detected or unsupported operations is requested
+we use the original pandas framework.
+
+Partitions
+----------
+
+Partitioning is used to achieve high parallelism. In the case of OmniSciDB
+based execution parallelism is provided by OmniSciDB execution engine
+and we don't need to manage multiple partitions.
+:py:class:`~modin.experimental.core.execution.native.implementations.omnisci_on_native.dataframe.dataframe.OmnisciOnNativeDataframe`
+always has a single partition.
+
+A partition holds data in either ``pandas.DataFrame`` or ``pyarrow.Table``
+format. ``pandas.DataFrame`` is preferred only when we detect unsupported
+data type and therefore have to use ``pandas`` framework for processing.
+In other cases ``pyarrow.Table`` format is preferred. Arrow tables can be
+zero-copy imported into OmniSciDB. A query execution result is also
+returned as an Arrow table.
+
+Data Ingress
+------------
+
+When a user imports data in Modin DataFrame (from a file or from some python
+object like array or dictionary) they invoke one of the ``modin.pandas.io`` 
+functions (to read data from a file) or use :py:class:`~modin.pandas.dataframe.DataFrame` constructor
+(to create a Frame from an iterable object). Both of the paths lead to the
+:py:class:`~modin.core.execution.dispatching.factories.dispatcher.FactoryDispatcher`
```

Comment:
why this link isn't actually a link in rendered? and there are a few others that should be links but they aren't

Reply:
`:py:class:~ClassName` can only reference sections that were created with `.. autoclass:: ClassName`, there's no section where we generate docs for `FactoryDispatcher`, so there couldn't be a link